dev-docs/services/ci/README.md

```diff
@@ -0,0 +1,244 @@
+# The CI Service in Hail Batch
+
+Hail Batch includes a CI service which has three functional purposes:
+
+- Runs tests against pull requests
+- Merges PRs to the `main` branch
+- Deploys services to the live infrastructure
+
+## Structure and Operation
+
+The CI service itself is deployed as a Kubernetes service in the Hail Batch cluster. See 
+[architecture diagram](../Hail%20Batch%20Architectural%20Diagram.png).
+
+As part of its configuration, CI must be configured with an access token allowing it to operate on 
+behalf of a github account called hail-ci-robot.
+
+### CI Update Loop
+
+The core CI update loop operates over a set of "watched branches" which each track three types of state flag which can 
+be marked as dirty:
+- The state of github ("`github_changed`")
+- The state of the deployed system ("`batch_changed`")
+- The state of the watched branch itself (for example - its current commit) ("`state_changed`")
+
+In hail-is, there is exactly one watched branch, which is `hail-is/hail:main`.
+
+At a configurable frequency, the CI service will mark every flag for each of its watched branches as dirty, which
+will trigger a re-evaluation of the system state.
+
+In hail-is, this polling happens every 5 minutes.
+
+### Github Configuration
+
+To make CI more responsive it can be configured to receive webhook event triggers from the github repository. 
```

Comment:
```suggestion
To make CI more responsive it is configured to receive webhook event triggers from the github repository. 
```

Reply:
Again, this is an accident of trying to be both a general purpose doc ("can be") vs how we use it ("is")